flake.nix

```diff
@@ -401,6 +401,18 @@
           };
         };
 
+      nixos-lib = import (nixpkgs + "/nixos/lib") { };
+
+      # https://nixos.org/manual/nixos/unstable/index.html#sec-calling-nixos-tests
+      runNixOSTestFor = system: test: nixos-lib.runTest {
+        imports = [ test ];
+        hostPkgs = nixpkgsFor.${system};
+        defaults = {
+          nixpkgs.pkgs = nixpkgsFor.${system};
```

Comment:
Tested with the assertion

```
          assertions = [ { assertion = config.nix.package.drvPath == self.packages.${system}.default.drvPath; message = "Unexpected nix.package"; } ];
```

and checked that the assertion runs. :heavy_check_mark:
I'm not adding the assertion, because we may want to write a test which uses an older nix daemon. In other words the validity of the assertion is somewhat coincidental.